app/routes/payWithInterledger.tsx

```diff
@@ -0,0 +1,215 @@
+import { conform, useForm } from '@conform-to/react'
+import { getFieldsetConstraint, parse } from '@conform-to/zod'
+import { type WalletAddress } from '@interledger/open-payments'
+import {
+  type ActionFunctionArgs,
+  json,
+  redirect,
+  type LoaderFunctionArgs
+} from '@remix-run/node'
+import { Form, useActionData, useLoaderData } from '@remix-run/react'
+import { z } from 'zod'
+import { Header } from '~/components/header'
+import Quote from '~/components/quoteDialog'
+import { Button } from '~/components/ui/button'
+import { Field } from '~/components/ui/form/form'
+import { PayWithInterledgerMark } from '~/components/ui/logo'
+import { useDialogContext } from '~/lib/context/dialog'
+import { fetchQuote, initializePayment } from '~/lib/open-payments.server'
+import { getValidWalletAddress } from '~/lib/validators.server'
+import { commitSession, destroySession, getSession } from '~/session'
+import { formatAmount } from '~/utils/helpers'
+
+export async function loader({ request }: LoaderFunctionArgs) {
+  const searchParams = new URL(request.url).searchParams
+  const isQuote = searchParams.get('quote') || false
+  const receiver = searchParams.get('receiver') || ''
```

Comment:
With the current flow, the user will fill in the form and try to send the money without knowing that the current receiver is not a valid wallet address. What do we think about checking the validity of the receiver (ensure it is a valid wallet address) and return an early error message, instead of showing the form with an invalid wallet address?